src/fault_injector/fault_types/track_speed_limit_fault.py

```diff
@@ -3,17 +3,53 @@
 
 from src.component import Component
 from src.fault_injector.fault_types.fault import Fault, FaultConfiguration
+from src.interlocking_component.route_controller import IInterlockingDisruptor
+from src.logger.logger import Logger
+from src.wrapper.simulation_object_updating_component import (
+    SimulationObjectUpdatingComponent,
+)
+from src.wrapper.simulation_objects import Track
 
 
 class TrackSpeedLimitFault(Fault):
     """A fault affecting the speed limit of tracks."""
 
+    configuration: "TrackSpeedLimitFaultConfiguration"
+    old_speed_limit: int
```

Comment:
```suggestion
    old_speed_limit: float
```
The speed limit is a float (m/s; see [TraCI-Docs](https://sumo.dlr.de/pydoc/traci._edge.html#EdgeDomain-setMaxSpeed))